xwiki-platform-distribution/xwiki-platform-distribution-flavor/xwiki-platform-distribution-flavor-test/xwiki-platform-distribution-flavor-test-ui/src/test/it/org/xwiki/test/ui/appwithinminutes/PageClassFieldTest.java

```diff
@@ -0,0 +1,264 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.test.ui.appwithinminutes;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.openqa.selenium.By;
+import org.openqa.selenium.Keys;
+import org.openqa.selenium.WebElement;
+import org.xwiki.appwithinminutes.test.po.ApplicationClassEditPage;
+import org.xwiki.appwithinminutes.test.po.SuggestClassFieldEditPane;
+import org.xwiki.test.ui.po.InlinePage;
+import org.xwiki.test.ui.po.SuggestInputElement;
+import org.xwiki.test.ui.po.SuggestInputElement.SuggestionElement;
+import org.xwiki.xclass.test.po.ClassSheetPage;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Special class editor tests that address only the Page class field type.
+ *
+ * @version $Id$
+ * @since 10.6
+ */
+public class PageClassFieldTest extends AbstractClassEditorTest
+{
+    @Before
+    @Override
+    public void setUp() throws Exception
+    {
+        String className = getTestClassName();
+        getUtil().deleteSpace(className);
+        getUtil().createPage(className, "pageclassfieldpage1", "Content", className + " Page 1");
+        getUtil().createPage(className, "pageclassfieldpage2", "Content", className + " Page 2");
+        getUtil().createPage(Arrays.asList(className, "space"), "pageclassfieldtesthome", "Content",
+            className + " TestHome");
+        getUtil().gotoPage(className, getTestMethodName(), "edit",
+            "editor=inline&template=AppWithinMinutes.ClassTemplate&title=" + getTestMethodName() + " Class");
+        editor = new ApplicationClassEditPage();
+    }
+
+    @Test
+    public void testSuggestions() throws Exception
+    {
+        String className = getTestClassName();
+        SuggestInputElement pagePicker = new SuggestClassFieldEditPane(editor.addField("Page").getName()).getPicker();
+
+        // Make sure the picker is ready.
+        pagePicker.click().waitForSuggestions();
```

Comment:
@mflorea You can actually reproduce the issue if you copy paste let's say `Sandbox Test Page 1
` in a page suggest input:
![bug-02](https://user-images.githubusercontent.com/2213999/42949199-870901be-8b71-11e8-8703-1c7f2dfbe6e6.gif)

It's a bit different with the functional tests because AFAIK there is no click performed on the input (like I did) but I think it's still the same problem. We couldn't see this problem before with the User class field because there wasn't enough users on the wiki which mean that they are all loaded by the first request.

What can be done is making the constructor of `SuggestInputElement` click on the input, wait for suggestions, then close the dropdown. Maybe a better solution would be to fix the selectize if there is a real issue. WDYT?